### ISSUE TITLE
How to overcome overfitting in pycaret framework

### ISSUE THREAD

**Issue by MislavSag** [pycaret/pycaret#363]
First, I like PyCaret approach to modeling. It is an AutoML framework but gives the researcher enough freedom to choose preprocessing steps. It is very simple to use.

My greatest concern with your and all other AutoML methods (e.g. H2o automl) is overfitting.

For example I have tried your (binary) classification module. I got very good results on the train set (accuracy 0.8) but bad results on unseen data (accuracy 0.5) for several best models. I am not sure what can be done to overcome this overfitting problem in your framework. The problem is that every model has its own arguments that can be changed. For example max depth in random forest and decision trees. But I can't set this parameter, which is expected, but then it remains the question of how could I approach this problem in pycaret?

I have a complex time series problem, but I think the problem is general and probably the most difficult to solve inside the AutoMl set of models.

**Comment by Yard1** [comment]
You can pass your own hyperparameter grids to tune_model - maybe that will help?

**Comment by clgarciga** [comment]
Is it possible that your good results on the training data is due to the fact that your data are time series but pycaret uses k-Fold CV to choose the best model rather than some sort of splitting method specifically for time series such as, for example, [sklearn.model_selection.TimeSeriesSplit](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.TimeSeriesSplit.html#sklearn.model_selection.TimeSeriesSplit)?

**Comment by MislavSag** [comment]
@Yard1 I have just discovered `custom_grid` argument is added in PyCaret 2! I believe I can use this option to tune models better.

@clgarciga , I am sure you are right. Time series split or even PurgedKfold split from mlfinlab package would be a better option, but I am afraid PyCaret doesn't support custom fold objects?

**Comment by pycaret** [comment]
@MislavSag Custom folds are not supported at this time but we are working towards adding that for the future release.

**Comment by pycaret** [comment]
@MislavSag 

This issue has been fixed in the `pycaret-nightly` release. Please install pycaret-nightly using `pip install pycaret-nightly`. Feel free to open a new issue, in case the problem persists. This fix will be published to PyPi as `pycaret==2.2.0` on October 28, 2020.
